src/app/core/services/data-set.service.ts

```diff
@@ -204,6 +222,14 @@ export class DatasetService {
 
     console.log(`[Dataset Service] Starting recording process: ${configuration.path}, Scale: ${configuration.timeScaleFormat}, Period: ${configuration.period}, Datapoints: ${newDataSourceConfig.maxDataPoints}`);
 
+    // Fetch historical data from REST feed to populate the dataset
+    this.fetchHistoryData(uuid).subscribe(success => {
+      if (success) {
+      } else {
+        console.warn(`[Dataset Service] Failed to load historical data for ${configuration.path}, continuing with real-time data only`);
+      }
+    });
+
     // Emit at a regular interval using the last value. We use this and not sampleTime() to make sure that if there is no new data, we still send the last know value. This is to prevent dataset blanks that look ugly on the chart
```

Comment:
Two points here:
1- We should not push the same data twice. If no data is coming it needs to be visible in the chart else we might send the same data very often and the user will be oblivious to the problem.

2- The plugin need to schedule the update rather then the client. If the plugin sends on a schedule, schedule to be determined by the data request, say every 1 seconds if you want 30 seconds of data, or maybe Avery 10 seconds for one hour, every minute for 8 hours etc, we should be fine to pass the update directly.